pkg/detector/detector.go

```diff
@@ -1103,9 +1103,16 @@ func (d *ResourceDetector) HandlePropagationPolicyDeletion(policyID string) erro
 		return err
 	}
 
-	cleanupMarksFunc := func(obj metav1.Object) {
+	cleanupMarksFunc := func(obj metav1.Object) bool {
+		// When cleaning up the marks, the resource template may have already been bound to a new propagationPolicy, causing the marks' values to change.
```

Comment:
@whitewindmills Here is the comment I added to explain why unexpected label values might occur. Do you think it's clear enough?